Mathlib/Topology/Algebra/ContinuousMonoidHom.lean

```diff
@@ -413,3 +415,280 @@ theorem locallyCompactSpace_of_hasBasis (V : ℕ → Set Y)
+    cases f
+    cases g
+    congr
+    exact MulEquiv.ext_iff.mpr (congrFun h₁)
+
+@[to_additive]
+instance : MulEquivClass (M ≃ₜ* N) M N where
+  map_mul f := f.map_mul'
+
+@[to_additive]
+instance : HomeomorphClass (M ≃ₜ* N) M N where
+  map_continuous f := f.continuous_toFun
+  inv_continuous f := f.continuous_invFun
+
+@[to_additive] -- shortcut instance that doesn't generate any subgoals
+instance : CoeFun (M ≃ₜ* N) fun _ ↦ M → N where
+  coe f := f
+
+/-- Two continuous multiplicative isomorphisms agree if they are defined by the
+same underlying function. -/
+@[to_additive (attr := ext)
+  "Two continuous additive isomorphisms agree if they are defined by the same underlying function."]
+theorem ext {f g : M ≃ₜ* N} (h : ∀ x, f x = g x) : f = g :=
+  DFunLike.ext f g h
+
+@[to_additive]
+protected theorem congr_arg {f : M ≃ₜ* N} {x x' : M} : x = x' → f x = f x' :=
+  DFunLike.congr_arg f
+
+@[to_additive]
+protected theorem congr_fun {f g : M ≃ₜ* N} (h : f = g) (x : M) : f x = g x :=
+  DFunLike.congr_fun h x
+
+@[to_additive (attr := simp)]
+theorem coe_mk (f : M ≃* N) (hf1 : Continuous f.toFun) (hf2 : Continuous f.invFun) :
+    (mk f hf1 hf2 : M → N) = f := rfl
+
+@[to_additive]
+theorem toEquiv_eq_coe (f : M ≃ₜ* N) : f.toEquiv = f :=
+  rfl
+
+/-- Makes a continuous multiplicative isomorphism from
+a homeomorphism which preserves multiplication. -/
+@[to_additive "Makes an continuous additive isomorphism from
+a homeomorphism which preserves addition."]
+def mk' (f : M ≃ₜ N) (h : ∀ x y, f (x * y) = f x * f y) : M ≃ₜ* N :=
+  ⟨⟨f.toEquiv,h⟩, f.continuous_toFun, f.continuous_invFun⟩
+
+end coe
+
+section bijective
+
+@[to_additive]
+protected theorem bijective (e : M ≃ₜ* N) : Function.Bijective e :=
+  EquivLike.bijective e
+
+@[to_additive]
+protected theorem injective (e : M ≃ₜ* N) : Function.Injective e :=
+  EquivLike.injective e
+
+@[to_additive]
+protected theorem surjective (e : M ≃ₜ* N) : Function.Surjective e :=
+  EquivLike.surjective e
+
+@[to_additive]
+theorem apply_eq_iff_eq (e : M ≃ₜ* N) {x y : M} : e x = e y ↔ x = y :=
+  e.injective.eq_iff
+
+end bijective
+
+section refl
+
+variable (M)
+
+/-- The identity map is a continuous multiplicative isomorphism. -/
+@[to_additive (attr := refl) "The identity map is a continuous additive isomorphism."]
+def refl : M ≃ₜ* M :=
+  { MulEquiv.refl _ with
+  continuous_toFun := by continuity
+  continuous_invFun := by continuity }
+
+@[to_additive]
+instance : Inhabited (M ≃ₜ* M) := ⟨ContinuousMulEquiv.refl M⟩
+
+@[to_additive (attr := simp)]
+theorem coe_refl : ↑(refl M) = id := rfl
+
+@[to_additive (attr := simp)]
+theorem refl_apply (m : M) : refl M m = m := rfl
+
+end refl
+
+section symm
+
+/-- The inverse of a ContinuousMulEquiv. -/
+@[to_additive "The inverse of a ContinuousAddEquiv."]
+def symm (cme : M ≃ₜ* N) : N ≃ₜ* M :=
+  { cme.toMulEquiv.symm with
+  continuous_toFun := cme.continuous_invFun
+  continuous_invFun := cme.continuous_toFun }
```

Comment:
Please setup `simps` for the new structure, search for `initialize_simps_projections` for examples.

Reply:
Sorry that I doesn't know how to do it, is adding `initialize_simps_projections ContinuousMulEquiv (toFun → apply, invFun → symm_apply)` after the def?

Reply:
This looks right to me (that's my understanding of the docstring for `initialize_simps_projections` at least)